SARS-CoV-2-Mpro/consolidate-work-to-pandas.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python
+
+"""
+Extract final work values to a single pandas dataframe
+"""
+
+import os
+
+projects = {
+    'complex' : 13418,
+    'solvent' : 13419,
+    }
+
+NUM_EXPECTED_WORK = 4001
+NUM_EXPECTED_WORK = 401
+NUM_EXPECTED_WORK = 41
+
+phases = list(projects.keys())
+
+basepaths = [
+    '/home/server/server2/data/SVR314342810/',
+    ]
+
+import datetime
+timestamp = datetime.datetime.now()
+
+def get_results(PROJ):
+    """
+    Get all result directories for a given project
+
+    Parameters
+    ----------
+    PROJ : str
+        PROJ, such as 'PROJ13400'
+
+    Returns
+    -------
+    results : dict
+        results[path] = (PROJ,RUN,CLONE,GEN) where path is the results directory
+    """
+    from tqdm import tqdm
+    
+    # Make a list of all results packets paths from all locations
+    paths = list()
+    from glob import glob # glob takes forever on S3 file gateway, so we don't use it
+    import os
+    import re
+    for basepath in basepaths:
+        #paths += glob(f'{basepath}/{PROJ}/RUN*/CLONE*/results*')
+        print(basepath)
+        for RUN in tqdm(os.listdir(f'{basepath}/{PROJ}')):
+            if not re.search('RUN', RUN): continue
+            for CLONE in os.listdir(f'{basepath}/{PROJ}/{RUN}'):
+                if not re.search('CLONE', CLONE): continue
+                for result in os.listdir(f'{basepath}/{PROJ}/{RUN}/{CLONE}'):
+                    if re.search('results\d+$', result):
+                        path = f'{basepath}/{PROJ}/{RUN}/{CLONE}/{result}'
+                        paths.append(path)
+
+    # Construct results
+    results = dict()
+    nfailures = 0
+    nsuccesses = 0
+    for path in tqdm(paths):
+        match = re.search('(?P<RUN>RUN\d+)/(?P<CLONE>CLONE\d+)/results(?P<GEN>\d+)$', path)
+
+        # Skip failures
+        if match is None:
+            nfailures += 1
+            continue
+        nsuccesses += 1
+        
+        RUN = match.group('RUN')
+        CLONE = match.group('CLONE')
+        GEN = f"{match.group('GEN')}"
+
+        results[path] = (PROJ,RUN,CLONE,GEN)
+
+    failure_rate = float(nfailures) / float(nfailures + nsuccesses)
+    print(f'{PROJ} : {nsuccesses} successes, {nfailures} failures ({failure_rate*100:.3}% failure rate)')
+        
+    return results
+            
+# Get all results directories
+print('Gathering all results packets...')
+results = dict()
+for phase in phases:
+    PROJ = f'PROJ{projects[phase]}'
+    print(PROJ)
+    results.update(get_results(PROJ))
+            
+def get_work(path):
```

Comment:
nit to go with nit below:
```suggestion
def get_work(path, params):
    PROJ, RUN, CLONE, GEN = params
```